src/sqlfluff/testing/rules.py

```diff
@@ -59,6 +59,9 @@ def assert_rule_fail_in_sql(code, sql, configs=None, line_numbers=None):
     linted = Linter(config=cfg).lint_string(sql, fix=True)
     lerrs = linted.get_violations()
     print("Errors Found: {0}".format(lerrs))
+    for e in lerrs:
+        if e.desc().startswith("Unexpected exception"):
+            pytest.fail(f"Linter failed with {e.desc()}")
```

Comment:
This is the line causing the coverage check to complain -- if we were hitting this line, that'd mean we have a failing test. I think having this line uncovered is fine.